src/qibolab/instruments/qm.py

```diff
@@ -110,99 +111,38 @@ def add(self, pulse):
         return qmpulse
 
 
-class QMOPX(AbstractInstrument):
-    """Instrument object for controlling Quantum Machines (QM) OPX controllers.
-
-    Playing pulses on QM controllers requires a ``config`` dictionary and a program
-    written in QUA language. The ``config`` file is generated in parts in the following places
-    in the ``register_*`` methods. The controllers, elements and pulses are all
-    registered after a pulse sequence is given, so that the config contains only
-    elements related to the participating qubits.
-    The QUA program for executing an arbitrary qibolab ``PulseSequence`` is written in
-    ``play`` and ``play_pulses`` and executed in ``execute_program``.
-
-    Args:
-        name (str): Name of the instrument instance.
-        address (str): IP address and port for connecting to the OPX instruments.
+@dataclass
+class QMConfig:
+    """Configuration for communicating with the ``QuantumMachinesManager``."""
 
-    Attributes:
-        is_connected (bool): Boolean that shows whether instruments are connected.
-        manager (:class:`qm.QuantumMachinesManager.QuantumMachinesManager`): Manager object
-            used for controlling the QM OPXs.
-        config (dict): Configuration dictionary required for pulse execution on the OPXs.
-        time_of_flight (int): Time of flight used for hardware signal integration.
-        smearing (int): Smearing used for hardware signal integration.
-    """
+    version: int = 1
+    controllers: dict = field(default_factory=dict)
```

Comment:
Of course, would be wonderful to have also these out of `dict`s, but one step at a time